app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -52,4 +76,66 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
             true
         }
     }
+
+    private fun getClickListenerForSyncNow(): Preference.OnPreferenceClickListener {
+        return Preference.OnPreferenceClickListener {
+            requireComponents.backgroundServices.syncManager.syncNow()
+            true
+        }
+    }
+
+    private val syncStatusObserver = object : SyncStatusObserver {
+        override fun onStarted() {
+            CoroutineScope(Dispatchers.Main).launch {
+                val pref = findPreference<Preference>(context?.getPreferenceKey(R.string.pref_key_sync_now))
+
+                pref.title = getString(R.string.sync_syncing)
+                pref.isEnabled = false
+            }
+        }
+
+        // Sync stopped successfully.
+        override fun onIdle() {
+            CoroutineScope(Dispatchers.Main).launch {
+                val pref = findPreference<Preference>(context?.getPreferenceKey(R.string.pref_key_sync_now))
+                pref.title = getString(R.string.preferences_sync_now)
+                pref.isEnabled = true
+                updateLastSyncedTimePref(context!!, pref, failed = false)
```

Comment:
Context might be null in some cases. It seems like we'd rather do nothing if there isn't a context. onIdle() especially seems likely to happen after we're no longer attached. Will this observer be cleaned up when the fragment is?

Reply:
Observer will be cleaned up, yes - the registration call includes an `owner` for that exact purpose. So once fragment is detached, these methods won't fire. so `context` _should_ always be present here.